### DISCUSSION TITLE
Add this operator to make static string methods into extension methods.

### DISCUSSION THREAD

**Discussion by JustinMinnaar** [dotnet/runtime#106441] [Ideas]
It would be nice if existing static methods on the String class could be changed to extension methods.

Then, instead of
```
   if (String.IsNullOrWhitespace(x)) ...
```

You would be able to use
```
    If (x.IsNullOrWhitespace()) ...
```

This could easily be done by modifying the string library and adding the this operator to the methods.

**Comment by huoyaoyuan** [comment]
As described in https://github.com/dotnet/csharplang/discussions/8363#discussioncomment-10345337 , this is not supported by C# language now.

**Comment by stephentoub** [comment]
https://github.com/dotnet/runtime/issues/79020

**Reply by daiplusplus** [reply]
>  We're not going to add IsNullXx extension methods as they're inherently confusing given that they syntactically look like they're calling a method on an instance that could be null.

A sound argument - but the ergonomics issue remains. What if we had another variation on the `.` operator (just like `?.` and `!.`) for this situation - which avoids the visual-ambiguity problem - something like `^.` or `/.` and used like so:

    ```
    maybeEmptyOrNullString^.IsNullOrWhitespace()

    maybeEmptyOrNullString/.IsNullOrWhitespace()
    ```

(I suggest `^.` or `/.` because they both _kiiinda_ resemble the delta symbol - as in Voltage _ΔV_ - and I bring that up that's because it's invoking [a _static_ member](https://en.wikipedia.org/wiki/Static_electricity) (_groan!_))

**Edit**: 2 downvotes already without any explanation? This community is worse than StackOverflow's...

**Reply by JustinMinnaar** [reply]
I think you may have nailed it. This could provide a shortcut to using the static method, without having to change the static methods that already exist. When typing, having referenced the variable and needing the static method, the IDE could offer the static methods and use the ^ (static) operator to access the method, or property, or field.
